src/NHibernate/Driver/SqlClientDriver.cs

```diff
@@ -144,6 +145,9 @@ protected static void SetDefaultParameterSize(DbParameter dbParam, SqlType sqlTy
 				case DbType.DateTimeOffset:
 					dbParam.Size = MaxDateTimeOffset;
 					break;
+				case DbType.Time:
```

Comment:
This should be in 2008 driver, as 2005 does not support time and reverts it to DateTime. Same for DateTime2.

Reply:
And same for `DateTimeOffset`. Defining defaults for those type supported only for 2008+ in the driver for previous versions is not very sound, excepted that the current pattern of sharing that method with SQL Server Ce, causing it to be static, causes it to be hard to be overridden in 2008 driver.

The reverting for `DbType.Time` and  `DbType.Date` seems to be a specific "feature" of `DbParameter` interface, when setting types through `DbType` property, forcing us to set them through `SqlDbType`, and only into 2008 driver.

To clean-up that maybe we should cease sharing with SQL Server CE, switch to instance methods and override them. But it would be a breaking change for those using `DateTime2` and `DateTimeOffset` with the old SQL Server driver.

Reply:
Another option could be to check the dialect in `SqlClientDriver` for flagging whether we are using 2008+ or not, and if yes port and use corresponding code from `SqlClient2008Driver`. That way, no need to override anything anymore in  `SqlClient2008Driver`, which would be then set as obsolete.